zerver/lib/webhooks/common.py

```diff
@@ -114,3 +130,38 @@ def validate_extract_webhook_http_header(request: HttpRequest, header: str,
         raise MissingHTTPEventHeader(header)
 
     return extracted_header
+
+class AuthStrategies:
+    SIMPLE_TOKEN = 1
+
+def http_api_callback(bot: UserProfile, auth_strategy: int,
+                      callback_url: str) -> Optional[Dict[str, Any]]:
+    """
+        The way we contact the callback endpoint will vary based on the
+        authentication method required to contact it.
+
+        For SIMPLE_TOKEN based auth, we can simply modify the url of the
+        endpoint accordingly.
+    """
+    try:
+        if int(auth_strategy) == AuthStrategies.SIMPLE_TOKEN:
+            bot_configuration = {config.key: config.value
+                                 for config in BotConfigData.objects.filter(bot_profile=bot)}
+            details = {
+                "callback_url": callback_url,
+                "auth_param": bot_configuration["callback_auth_param"],
+                "auth_token": bot_configuration["callback_auth_token"],
+            }
+            complete_url = "{callback_url}?{auth_param}={auth_token}".format(**details)
+            response = requests.get(complete_url)
+            if response.status_code != 200:
+                raise UnsuccessfulResponseError(bot, callback_url, response.status_code)
+            else:
+                return json.loads(response.text)
+        else:
+            raise ValueError("Unsupported auth_strategy.")
```

Comment:
So, a couple of things here:
* Usually, the `try` block needs to be as small as possible. The one above contains a bit too much code, so I would recommend cleaning this up a bit and to properly isolate the line that you think could potentially raise a `KeyError`.
* Also, it is probably a better approach to have separate handler functions for different auth strategies, and such handlers should do their own specific error handling. :)

Reply:
Both are good ideas! I've made the changes.